event-sourcing/event-sourcing-core/src/main/scala/org/apache/james/eventsourcing/CommandDispatcher.scala

```diff
@@ -65,28 +79,28 @@ class CommandDispatcher @Inject()(eventBus: EventBus, handlers: Set[CommandHandl
       .map(_ => ())
 
 
-  private def tryDispatch(c: Command): Boolean = {
-    val maybeEvents: Option[Try[List[_ <: Event]]] = handleCommand(c)
-    maybeEvents match {
-      case Some(eventsTry) =>
-        eventsTry
-          .flatMap(events => Try(eventBus.publish(events))) match {
-          case Success(_) => true
-          case Failure(_: EventStoreFailedException) => false
-          case Failure(e) => throw e
-        }
+  private def tryDispatch(c: Command): SMono[Boolean] = {
+    handleCommand(c) match {
+      case Some(eventsPublisher) =>
+        SFlux(eventsPublisher)
+          .collectSeq()
+          .flatMap(events => eventBus.publish(events)
+            .`then`(SMono.just(true))
+            .onErrorResume {
+              case _: EventStoreFailedException => SMono.just(false)
+              case e => SMono.raiseError(e)
+            })
       case _ =>
         throw CommandDispatcher.UnknownCommandException(c)
     }
   }
 
-  private def handleCommand(c: Command): Option[Try[List[_ <: Event]]] = {
+  private def handleCommand(c: Command): Option[Publisher[_ <: Event]] = {
```

Comment:
`Option[Publisher]` looks weird :)

Reply:
Sure, but you might not get one.

Reply:
I meant, why not `Publisher[_ <: Event]` ?

Reply:
Because you have to make the distinction between an empty `Publisher` having no `Event` and no `Publisher`